test/factories/host_related.rb

```diff
@@ -299,6 +299,17 @@ def set_nic_attributes(host, attributes, evaluator)
       subnet
     end
 
+    trait :with_rootpass do
+      root_pass 'xybxa6JUkz63w'
+    end
+
+    trait :with_os do
```

Comment:
I'd suggest something like `managed` to match the host, since this does more than add just the operatingsystem association, which can cause problems when we come to use it later.